src/frontends/lean/inductive_cmds.cpp

```diff
@@ -693,6 +697,61 @@ class inductive_cmd_fn {
         parse_mutual_inductive(params, inds, intro_rules);
         return shared_inductive_cmd(params, inds, intro_rules);
     }
+
+    environment shared_coinductive_cmd(buffer<expr> const & params,
+                                       buffer<expr> const & inds,
+                                       buffer<buffer<expr> > const & intro_rules) {
+        buffer<expr> new_params;
+        buffer<expr> new_inds;
+        buffer<buffer<expr> > new_intro_rules;
+        if (!m_explicit_levels) {
+            m_explicit_levels = has_explicit_level(intro_rules);
+        }
+        elaborate_inductive_decls(params, inds, intro_rules, new_params, new_inds, new_intro_rules);
+
+        expr first_ind = inds[0];
+        metavar_context mctx;
+        options opts = m_p.get_options();
+
+        vm_state vms(m_env, opts);
+        scope_vm_state vms_scope(vms);
+        tactic_state s = mk_tactic_state_for(m_env, opts, mlocal_name(first_ind), mctx, local_context(), mk_true());
+        buffer<vm_obj> args;
+        buffer<vm_obj> inds_arg;
+        for (unsigned i = 0; i < new_inds.size(); i ++) {
+            inds_arg.push_back(mk_vm_pair(to_obj(new_inds[i]), to_obj(new_intro_rules[i])));
+        }
+        args.push_back(to_obj(m_lp_names));
+        args.push_back(to_obj(new_params));
+        args.push_back(to_obj(inds_arg));
+
+        vm_obj r = tactic_evaluator(m_ctx, opts, mk_true())
+            (mk_constant(get_tactic_add_coinductive_predicate_name()), args, s);
```

Comment:
Then I will use `first_ind` to return errors. I guess this will report the error at the first identifier. Or is it possible to encode the beginning of the `coinductive` command as `expr `?